packages/dds/tree/src/core/edit-manager/editManager.ts

```diff
@@ -3,84 +3,131 @@
  * Licensed under the MIT License.
  */
 
+import BTree from "sorted-btree";
 import { assert } from "@fluidframework/common-utils";
 import { ChangeFamily } from "../change-family";
-import { TaggedChange, RevisionTag, tagChange, tagInverse } from "../rebase";
 import { SimpleDependee } from "../dependency-tracking";
 import { AnchorSet, Delta, emptyDelta } from "../tree";
-import { brand, Brand, fail, getOrCreate, RecursiveReadonly } from "../../util";
-
-export interface Commit<TChangeset> {
-	sessionId: SessionId;
-	seqNumber: SeqNumber;
-	refNumber: SeqNumber;
-	changeset: TChangeset;
-}
-
+import {
+	brand,
+	Brand,
+	fail,
+	getOrCreate,
+	mapIterable,
+	Mutable,
+	RecursiveReadonly,
+	zipIterables,
+} from "../../util";
+import {
+	findAncestor,
+	findCommonAncestor,
+	mintCommit,
+	GraphCommit,
+	RevisionTag,
+	SessionId,
+	Rebaser,
+	assertIsRevisionTag,
+} from "../rebase";
+
+export interface Commit<TChangeset> extends Omit<GraphCommit<TChangeset>, "parent"> {}
 export type SeqNumber = Brand<number, "edit-manager.SeqNumber">;
-export type SessionId = string;
+export const minimumPossibleSequenceNumber: SeqNumber = brand(Number.MIN_SAFE_INTEGER);
+
+const nullRevisionTag = assertIsRevisionTag("00000000-0000-4000-8000-000000000000");
 
 /**
  * Represents a local branch of a document and interprets the effect on the document of adding sequenced changes,
  * which were based on a given session's branch, to the document history
  */
-// TODO: Remove commits when they are no longer in the collab window
 // TODO: Try to reduce this to a single type parameter
-// TODO: Move logic into Rebaser if possible
 export class EditManager<
 	TChangeset,
 	TChangeFamily extends ChangeFamily<any, TChangeset>,
 > extends SimpleDependee {
-	// The trunk represents the list of received sequenced changes.
-	// The change in each commit is rebased onto the previous change in the list.
-	private readonly trunk: Commit<TChangeset>[] = [];
+	/** The trunk represents the list of received sequenced changes. */
```

Comment:
Our [documentation guidlines](https://github.com/microsoft/FluidFramework/wiki/TSDoc-Guidelines) stipulate that the multi-line form should always be used. I'm not sure why though.

Reply:
It would be nice to mention that this is the head of the trunk.

Reply:
Good call